clang/lib/CodeGen/BackendUtil.cpp

```diff
@@ -98,13 +99,16 @@ extern cl::opt<bool> PrintPipelinePasses;
 static cl::opt<bool> ClSanitizeOnOptimizerEarlyEP(
     "sanitizer-early-opt-ep", cl::Optional,
     cl::desc("Insert sanitizers on OptimizerEarlyEP."), cl::init(false));
-}
+
+extern cl::opt<InstrProfCorrelator::ProfCorrelatorKind> ProfileCorrelate;
```

Comment:
Can be extracted out as refactoring (without adding the new correlation kind).

Reply:
yes - just use the debug-info kind and non-kind.

Reply:
There are three options to ProfileCorrelate, none (default), debug-info and binary. This patch introduces the binary mode, which is controlled by -mllvm -profile-correlate=binary. And both binary and debug-info correlation will have the default profile name `default_%m.proflite`.

Reply:
right -- what I meant is that you can move this option into the refactor patch. In that patch, the option will only have two options.

Reply:
Oh, I see. Do you mean remove ProfCorrelatorKind::None? Just one for debug-info and another one for binary?

Reply:
No, what I meant is in the refactoring/preparation patch, change debug-info-correlate into profile-correlate=debug. The none kind is still kept. 

In your next patch to implement binary correlation, add the binary kind.

The goal for the preparation/refactoring patch is to minimize the diff in the second patch implementing the feature so that reviewer can have more focus on the functional diffs.